manifests/agent/follow.pp

```diff
@@ -39,11 +40,15 @@
     } else {
       fail 'can not specify both, token and destination'
     }
+    
+    if $multi_line_start_pattern != '' {
+      $entry_identifier = "entry_identifier = ${multi_line_start_pattern}\n"
+    }
 
     file { "${confd_path}.conf":
-      content => "# managed by puppet, module ${::module}\n[${clean_title}]\npath = ${my_path}\n${token_or_destination}\n",
+      content => "# managed by puppet, module ${::module}\n[${clean_title}]\npath = ${my_path}\n${token_or_destination}\n${$entry_identifier}",
```

Comment:
@gearoidof, moved the new line, drawback is that when not multi-line we have 1 empty line on the generated file.
I hope you are able to merge this now.